apiserver/charms_test.go

```diff
@@ -77,18 +77,24 @@ func (s *charmsSuite) TestAuthRequiresUser(c *gc.C) {
 	err = machine.SetPassword(password)
 	c.Assert(err, jc.ErrorIsNil)
 
-	resp, err := s.sendRequest(c, machine.Tag().String(), password, "POST", s.charmsURI(c, ""), "", nil)
+	resp, err := s.sendRequest(c, httpRequestParams{
+		tag:      machine.Tag().String(),
+		password: password,
+		method:   "POST",
+		url:      s.charmsURI(c, ""),
+		nonce:    "fake_nonce",
+	})
 	c.Assert(err, jc.ErrorIsNil)
-	s.assertErrorResponse(c, resp, http.StatusUnauthorized, "unauthorized")
```

Comment:
Note that the actual error here (discarded by the API) as "machine 0 not provisioned" because the sent request did not contain a machine nonce.